game/views.py

```
@@ -496,43 +496,46 @@ def check_username(request):
     if not username:
         return JsonResponse({'available': False, 'error': 'No username provided'}, status=400)
-    exists = User.objects.filter(username__iexact=username).exists()
+    users = User.objects.filter(username__iexact=username)
+    pending_user_id = request.session.get('registration_user_id')
+    if pending_user_id:
+        users = users.exclude(id=pending_user_id, is_active=False)
+    exists = users.exists()
     return JsonResponse({'available': not exists})
 
 
+def _pending_user_matches_signup(pending_user, username, email):
```

Ruff (0.15.14)

[warning] Missing return type annotation for private function `_pending_user_matches_signup`

(ANN202)